src/System.Linq/src/System/Linq/Select.SpeedOpt.cs

```diff
@@ -157,6 +157,127 @@ public TResult TryGetLast(out bool found)
             }
         }
 
+        private sealed partial class SelectRangeIterator<TResult> : Iterator<TResult>, IPartition<TResult>
+        {
+            private readonly int _start;
+            private readonly int _end;
+            private readonly Func<int, TResult> _selector;
+
+            public SelectRangeIterator(int start, int end, Func<int, TResult> selector)
+            {
+                Debug.Assert(start < end);
+                Debug.Assert((end - start) <= int.MaxValue);
+                Debug.Assert(selector != null);
+
+                _start = start;
+                _end = end;
+                _selector = selector;
+            }
+
+            public override Iterator<TResult> Clone() =>
+                new SelectRangeIterator<TResult>(_start, _end, _selector);
+
+            public override bool MoveNext()
+            {
+                if (_state < 1 || _state == (_end - _start + 1))
+                {
+                    Dispose();
+                    return false;
+                }
+
+                int index = _state++ - 1;
```

Comment:
nit: `state++` on a separate line for readability

(doesn't change assembly size: [sharplab.io](https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQxNMAfAAgBgARYCMA3ALABQWAzAQEx4DCeA3hXuwTQJYB2MeAWUIAKXvwD6sCPACULNh0Vi8vACZwAHngC8eSTGlwA1EbwBaPCQWL2WAOx6p8PKbWay5RQF9reX9RU+QVpRIP1DOVZPG3ZlNy1dcOcLK2iYpOMjDxjbBwyXQPUNbI4fci8gA===))

Reply:
This is the same code that exists for the other like operators, as this was mostly copy-and-paste.  I'm going to leave it as is.  If it's important to you, please feel free to open a PR that addresses all of them.